bin/propolis-standalone/src/main.rs

```diff
@@ -1058,6 +1070,6 @@ fn main() -> anyhow::Result<()> {
     );
 
     // wait for instance to be destroyed
-    inst.wait_destroyed();
-    Ok(())
+    let code = inst.wait_destroyed();
+    std::process::exit(code);
```

Comment:
Does the move to an explicit exit here change the behaviour, because we are now not actually returning from **main()** any more?  In particular, I wonder if we previously would have called a bunch of **Drop** routines and now we're not doing that...

Reply:
Unless propolis components are spawning tasks which aren't themselves collected on instance destroy, we should be pretty well cleaned up by now.

That said, I've refactored and cleaned up some other bits so we can return an `ExitCode`.